src/admin/shared/services/api-service/api.service.ts

```diff
@@ -0,0 +1,32 @@
+import ky from 'ky-universal';
+import { KyInstance } from 'ky/distribution/types/ky';
+
+import { getEnv } from '../../../../shared/helpers';
+import { goToLoginBecauseOfUnauthorizedError } from '../../../../shared/helpers/fetch-with-logout';
+
+export abstract class ApiService {
+	private static api: KyInstance | null = null;
+
+	public static getApi(ignoreAuthError = false): KyInstance {
+		if (!ApiService.api) {
+			this.api = ky.create({
+				prefixUrl: getEnv('PROXY_URL'),
+				headers: {
+					'content-type': 'application/json',
+				},
+				credentials: 'include', // TODO change to same-origin once working on server
+				hooks: {
+					afterResponse: [
+						(_request, _options, response) => {
+							if (response.status === 401 && !ignoreAuthError) {
+								// User is no longer logged in => force them to login again
+								goToLoginBecauseOfUnauthorizedError();
+							}
+						},
+					],
+				},
```

Comment:
Does this conflict with any existing mechanisms (i.e. `check-login`) or make them obsolete?

Reply:
in principle, we'd like to use this service for all network calls.
Currently the dataService and fetchWithLogout function do something similar.